flow/aria.js

```diff
@@ -308,12 +308,24 @@ type ARIARoleRelation = {
 type ARIARoleRelationConcept = {
   name: string,
   attributes?: Array<ARIARoleRelationConceptAttribute>,
+  // Make these explicit. These values represent quirks of the mapping betweent
+  // ARIA and other semantic ontological systems.
+  constraints?: Array<
+    'direct descendant of document'
+    | 'direct descendant of ol, ul or menu'
+    | 'direct descendant of details element with the open attribute defined'
+    | 'descendant of table'
+  >,
 };
 
 type ARIARoleRelationConceptAttribute = {
   name: string,
   value?: string | number,
   // Make these explicit. These values represent quirks of the mapping betweent
   // ARIA and other semantic ontological systems.
-  constraints?: Array<'unset' | 'isset' | '>1'>,
+  constraints?: Array<
+    'undefined' // The attribute does not exist on the node: <a>
+    | 'set' // The attribute has a value: <a b="c">
```

Comment:
This is confusing to me. It sounds like this should not match empty attribute values e.g. `<input type="text" list="" />` should not be considered a combobox since the `list` attribute has no value.

On the other hand `set` would imply to me that the attribute is simply "set". It doesn't make a statement about the type of the value e.g. does it have any value or a certain value etc.

Compare this to a `<select />` element and its `multiple` attribute. The wording is the same `select (with a multiple [...]` vs `input type=text [...] with a list attribute` but for combobox we constrain the attribute to `set` and leave the `multiple` attribute unconstrained for `listbox`.

The current implementation recognizes `<select multiple="" />` as a `listbox` but `<input type="text" list="" />` is not considered a `combobox`.

I could not find statements about existence of a value in html-aria. Is it fair to assume that these constraints should be ignored when mapping html to aria (since it contains a comment about quirks in other systems)?